### DISCUSSION TITLE
Can't Update Many-to-One Relation Unique ID

### DISCUSSION THREAD

**Discussion by DavianYang** [prisma/prisma#12450] [Q&A] [answered]
hello, i have trouble updating unique id `countryId` in many-to-one relation `Country <- Address[]`.  Is there a way to update through `prisma.address.update()`

```prisma
model Address {
  id                     String    @id @default(uuid())
  createdAt              DateTime  @default(now())
  updatedAt              DateTime  @updatedAt
  fullName               String    @default("")
  country                Country   @relation(fields: [countryId], references: [id], onUpdate: Cascade)
  countryId              String    @unique
}
```

```prisma
model Country {
  id        String    @id @default(uuid())
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  name      String
  code      String    @unique @db.VarChar(255)
  address   Address[]
}
```

```ts
await this.prisma.address.update({
	where: { id: addressId }, 
	data: { 
		country: { 
			connect: {
					id: newCountryId
			}	
	}}
})
```

Error: `"\nInvalid prisma.country.update() invocation:\n\n\n  Unique constraint failed on the fields: (countryId)"`

**Comment by austincrim** [comment, marked as answer]
Hey @DavianYang 👋 ,

This query should work fine as long as you have an existing `country` with the given `newCountryId`. If you are not guaranteed to have an existing `country`, you could look into the [`connectOrCreate`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connectorcreate) API.

Are you certain that you are connect the `address` to a valid, existing `country`?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
